crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -100,6 +100,52 @@ fn given_hardcoded_contract_run_invoke_tx_then_it_works() {
     });
 }
 
+#[test]
+fn given_hardcoded_contract_run_invoke_tx_v0_then_it_works() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+
+        let mut transaction = get_invoke_dummy(Starknet::chain_id(), NONCE_ZERO);
```

Comment:
`get_invoke_dummy` will always return a TxV1. That what it is coded to do

Reply:
so you never enter `if let starknet_api::transaction::InvokeTransaction::V0(tx)`
and therefore you are execution a tx v1 here: `assert_ok!(Starknet::invoke(RuntimeOrigin::none(), transaction));`

Reply:
I'm resolving it now